Tests/Unit/Code/ScriptTagBuilderTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Tests\Unit\Code;
+
+use Brotkrueml\MatomoIntegration\Code\ScriptTagBuilder;
+use Brotkrueml\MatomoIntegration\Event\EnrichScriptTagEvent;
+use PHPUnit\Framework\MockObject\Stub;
+use PHPUnit\Framework\TestCase;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Core\Site\Entity\Site;
+
+/**
+ * This test check if the ScriptTagBuilder works as expected
+ */
+final class ScriptTagBuilderTest extends TestCase
+{
+    /**
+     * @var Stub|Site
+     */
+    private $siteStub;
+
+    /**
+     * @var Stub|ServerRequestInterface
+     */
+    private $requestStub;
+
+    protected function setUp(): void
+    {
+        $this->siteStub = $this->createStub(Site::class);
+
+        $this->requestStub = $this->createStub(ServerRequestInterface::class);
+        $this->requestStub
+            ->method('getAttribute')
+            ->with('site')
+            ->willReturn($this->siteStub);
+    }
+
+    /**
+     * @test
+     */
+    public function scriptTagWithoutAttributesRendered(): void
+    {
+        $enrichEvent = new EnrichScriptTagEvent($this->requestStub);
+
+        $eventDispatcherStub = $this->createStub(EventDispatcherInterface::class);
+        $eventDispatcherStub
+            ->method('dispatch')
+            ->willReturn($enrichEvent);
+
+        $scriptTagBuilder = new ScriptTagBuilder(
+            $eventDispatcherStub
+        );
+        $scriptTagBuilder->setServerRequest($this->requestStub);
+
+        self::assertSame(
+            '<script>/* some tracking code */</script>',
+            $scriptTagBuilder->build('/* some tracking code */')
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function enrichScriptTagViaEventAddsIdTypeAndDataAttributes(): void
+    {
+        $enrichEvent = new EnrichScriptTagEvent($this->requestStub);
```

Comment:
Use a data provider (so this test and the next can be combined).
See for example: https://github.com/brotkrueml/typo3-matomo-integration/blob/main/Tests/Unit/Code/JavaScriptTrackingCodeBuilderTest.php#L130
I also prefer one single test case for one check:
- When id is set to non-empty value
- When type is set to non-empty value
- When data-attribute with non-empty value is set
- When data-attribute with empty value is set
- When multiple data-attributes are set
- When all attributes are set to non-empty values

This way it is easier to spot errors.

Important are also tests to check special chars in values, like "<>&

Add also tests to check the exceptions:
Example: https://github.com/brotkrueml/typo3-matomo-integration/blob/main/Tests/Unit/Entity/CustomDimensionTest.php#L23